### ISSUE TITLE
OPTICS: Accomodate Precomputed Distance Matrix

### ISSUE THREAD

**Issue by maxcw** [annoviko/pyclustering#412]
It would be great to have the option of feeding a pre-computed distance matrix to OPTICS, since my data points do not have coordinates. This function is available in sklearn's dbscan, and would be very useful.

**Comment by annoviko** [comment]
Hello, @maxcw,

Thank you for your proposal, it's a good feature, I will implement it for the OPTICS algorithm.

**Comment by annoviko** [comment]
@maxcw, distance matrix as an input data is supported for OPTICS and DBSCAN. Currently it is available on branch '0.8.dev' and 'master'. You can obtain changes by cloning:
```bash
mkdir pyclustering
cd pyclustering
git clone https://github.com/annoviko/pyclustering.git .
```

After that you need to compile core of the library:
```bash
cd ccore
make ccore_x64 # if you are using 64-bit python
make ccore_x86 # if you are using 32-bit python
```

The changes will be available in the next release of pyclustering library - 0.8.1.
